### PR TITLE
Feat/logging: simple logging with verbose mode

### PR DESCRIPTION
**Pull request by jo-bru**
For logging, I used the [easylogging++](https://github.com/amrayn/easyloggingpp) library as it's quite simple and supports verbose logging.
Info, debug and evaluation messages can be logged by using the verbose levels (1,2,3) respectively. The verbosity level can be set by using the -v flag (e.g. `xarm-commander -vv get_state` => has verbosity level 2 = debug).
The response code and response values are returned with `std::cout` formatted as JSON (maybe this should be done by using an JSON support library).

I'm not sure if using the verbose levels is the way to go, or if we need to use the INFO, DEBUG, etc. levels and implement the verbosity ourselves.

I was trying to redirect the output of the xarm-SDK to the logger (#5), but since it uses `printf()` instead of `std::cout` it's not trivial and must be solved by using pipes, I guess (which I do not understand yet).

This first logging implementation should solve #4, but I guess we'll have to rethink the approach after first integration tests with nodejs.

### CODE REVIEW

**Comment by jo-bru** [comment]
Sorry for the mess in the commit history:grimacing:

**Comment by iomz** [comment]
Just before reviewing the code
>I was trying to redirect the output of the xarm-SDK to the logger (#5), but since it uses printf() instead of std::cout it's not trivial and must be solved by using pipes, I guess (which I do not understand yet).

How about we tweak the SDK and suppress/manipulate the nasty messages from the SDK? I forked the repo: https://github.com/Interactions-HSG/xArm-CPLUS-SDK

File: src/xarm-commander.cpp [hunk 1]
```diff
@@ -1,40 +1,69 @@
...
+constexpr int kLogEval = 3;
+}  // namespace constants
+
+INITIALIZE_EASYLOGGINGPP
+
+auto InitXarm(const std::string &port) -> XArmAPI * {
+    // TODO(jo-bru): redirect stdout to logger (using pipes?)
+    auto *arm = new XArmAPI(port);
+    return arm;
+}
```
Comment (iomz) [inline src/xarm-commander.cpp]: Assignment to this (kind of) global `arm` pointer may be redundant. Let's discuss it when we create the `server` command.

**Review by iomz — APPROVED** [review]
Thanks, it's getting cleaner every week 👍 
Please review #24 before merging this to `main`.

Also, we'd better take into account the information loss between the `XArmAPI::` wrappers and the core implementation invocations. (For instance, the ip/port problem in 5e75167d4b683f472f98e10d514038d18fdb629f)

Finally, let's not use actual values as defaults 😄

**Review by iomz — COMMENTED** [review]
By the way, I haven't figured out how to correctly set a custom TCP port through `XArmAPI()` 😓 
Perhaps. I can open the issue and address this inconsistency in their repo.

**Comment by jo-bru** [comment]
> How about we tweak the SDK and suppress/manipulate the nasty messages from the SDK? I forked the repo: https://github.com/Interactions-HSG/xArm-CPLUS-SDK
Ah yes, that would be easier..

**Comment by jo-bru** [comment]
> Finally, let's not use actual values as defaults 😄

What do you mean by that?

**Comment by iomz** [comment]
Ah no, it's about 130.82.171.9😉

**Pull request merged**